src/classes/Metrics.js

```diff
@@ -209,6 +209,7 @@ class Metrics {
 				this._recordActive();
 				break;
 			case 'engaged':
+				this._recordEngagedWithRepeats();
```

Comment:
With the refactor of the way repeats are tracked, I think you can consolidate these functions to avoid the calculashun duplicashun